src/main/java/org/mockito/ArgumentCaptor.java

```diff
@@ -100,12 +84,41 @@ private ArgumentCaptor(Class<? extends T> clazz) {
      * @return null or default values
      */
     public T capture() {
-        Mockito.argThat(capturingMatcher);
+        return captureIf(Any.ANY);
```

Comment:
The question is what dit you expect as a user in the above described scenario? What type's  should `ArgumentCaptor.forClass(String.class)` capture:
 - String ?
 - null ?
 - Integer ?

At least I would not expect that an Integer is captured, if not why? If you capture Integers how would you get the values? Lets look a that example:

```
ArgumentCaptor<String> argumentCaptor = ArgumentCaptor.forClass(String.class);
mock.forObject("capturedValue");
mock.forObject(42);
verify(mock,times(2)).forObject(argumentCaptor.capture());//No TooManyActualInvocations-Exeception this time

String lastArg= argumentCaptor.getValue();//ClassCastException cant cast a Integer to String
```